skxray/core/speckle_fitting.py

```diff
@@ -0,0 +1,175 @@
+# STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OTHERWISE) ARISING   #
+# IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE   #
+# POSSIBILITY OF SUCH DAMAGE.                                          #
+########################################################################
+
+"""
+    This module will provide fitting tools for
+    X-ray Speckle Visibility Spectroscopy (XSVS)
+"""
+
+from __future__ import (absolute_import, division, print_function,
+                        unicode_literals)
+import six
+
+from six.moves import zip
+from six import string_types
+
+import logging
+
+import numpy as np
+from time import time
+
+from scipy.stats import nbinom
+from scipy.special import gamma, gammaln
+
+logger = logging.getLogger(__name__)
+
+
+def negative_binom_distribution(bin_edges, K, M):
+    """
+    Negative Binomial (Poisson-Gamma) distribution function
+
+    Parameters
+    ----------
+    bin_edges : array
+        normalized speckle count bin centers
+
+    K : int
+        number of photons
+
+    M : int
+        number of coherent modes
+
+    Returns
+    -------
+    poisson_gamma : array
+        Negative Binomial (Poisson-Gamma) distribution function
+
+    Notes
+    -----
+    The negative-binomial distribution function
+
+    :math ::
+        P(K) = \frac{\\Gamma(K + M)} {\\Gamma(K + 1) ||Gamma(M)}(\frac {M} {M + <K>})^M (\frac {<K>}{M + <K>})^K
+
+    These implementation is based on following references
+    References: text [1]_, text [2]_
+
+    .. [1] L. Li, P. Kwasniewski, D. Oris, L Wiegart, L. Cristofolini,
+       C. Carona and A. Fluerasu , "Photon statistics and speckle visibility
+       spectroscopy with partially coherent x-rays" J. Synchrotron Rad.,
+       vol 21, p 1288-1295, 2014.
+
+    .. [2] R. Bandyopadhyay, A. S. Gittings, S. S. Suh, P.K. Dixon and
+       D.J. Durian "Speckle-visibilty Spectroscopy: A tool to study
+       time-varying dynamics" Rev. Sci. Instrum. vol 76, p  093110, 2005.
+
+    """
+    co_eff = np.exp(gammaln(bin_edges + M) -
+                    gammaln(bin_edges + 1) - gammaln(M))
+
+    poisson_gamma = co_eff * np.power(M / (K + M), M) * np.power(K / (M + K),
+                                                                 bin_edges)
+    return poisson_gamma
+
+
+def poisson_distribution(bin_edges, K):
+    """
+    Poisson Distribution
+
+    Parameters
+    ---------
+    K : int
+        number of photons
+
+    bin_edges : array
+        normalized speckle count bin centers
+
+    Returns
+    -------
+    poisson_dist : array
+       Poisson Distribution
+
+    Notes
+    -----
+    These implementations are based on the references under
+    nbinom_distribution() function Notes
+
+    :math ::
+        P(K) = \frac{<K>^K}{K!}\exp(-K)
+    """
+    poisson_dist = np.exp(-K) * np.power(K, bin_edges)/gamma(bin_edges + 1)
+    return poisson_dist
+
+
+def gamma_distribution(bin_edges, K, M):
```

Comment:
Does this do anything that `scipy.stats.gamma` does not do?

http://docs.scipy.org/doc/scipy-0.15.1/reference/generated/scipy.stats.gamma.html